engines/sci/graphics/frameout.cpp

```diff
@@ -55,31 +57,37 @@
 
 namespace Sci {
 
-GfxFrameout::GfxFrameout(SegManager *segMan, ResourceManager *resMan, GfxCoordAdjuster *coordAdjuster, GfxScreen *screen, GfxPalette32 *palette, GfxTransitions32 *transitions) :
+GfxFrameout::GfxFrameout(SegManager *segMan, ResourceManager *resMan, GfxCoordAdjuster *coordAdjuster, GfxPalette32 *palette, GfxTransitions32 *transitions, GfxCursor32 *cursor) :
 	_isHiRes(ConfMan.getBool("enable_high_resolution_graphics")),
 	_palette(palette),
+	_cursor(cursor),
 	_resMan(resMan),
-	_screen(screen),
 	_segMan(segMan),
 	_transitions(transitions),
 	_benchmarkingFinished(false),
 	_throttleFrameOut(true),
 	_throttleState(0),
-	// TODO: Stop using _gfxScreen
-	_currentBuffer(screen->getDisplayWidth(), screen->getDisplayHeight(), nullptr),
 	_remapOccurred(false),
 	_frameNowVisible(false),
-	_screenRect(screen->getDisplayWidth(), screen->getDisplayHeight()),
 	_overdrawThreshold(0),
 	_palMorphIsOn(false) {
 
-	_currentBuffer.setPixels(calloc(1, screen->getDisplayWidth() * screen->getDisplayHeight()));
-
-	// QFG4 is the only SCI32 game that doesn't have a high-resolution toggle
+	// QFG4 is the only SCI32 game that doesn't have a high-resolution version
 	if (g_sci->getGameId() == GID_QFG4) {
 		_isHiRes = false;
 	}
 
+	if (g_sci->getGameId() == GID_PHANTASMAGORIA) {
+		_currentBuffer = Buffer(630, 450, nullptr);
+	} else if (_isHiRes) {
+		_currentBuffer = Buffer(640, 480, nullptr);
+	} else {
+		_currentBuffer = Buffer(320, 200, nullptr);
+	}
```

Comment:
> This is about keeping existing code behavior and not just removing it without any reason to do so.

All SCI32 graphics code was guesswork and hacks before I started working on it. I am not sure why any of the SCI32 Mac code would be exempt from this fact. In fact… (read next response)

> We already know that GK1 on Mac seems to be hires only, so why not just keep that information inside the code instead of removing it now and having to add it again in some months.

The default value for `enable_high_resolution_graphics` is true. So GK1 Mac will already be hires without adding a single line of special code to support it.

> Otherwise we could also just remove any Macintosh code altogether including all SCI32 Mac detection entries […]

This is a false equivalence. Detection table hashes are data, not code, and weren’t based on guesswork.

> And well, I could also remove Hoyle1+2+SQ3 Mac detection entries too, because those are also not working correctly atm and I had not enough nerve to try to figure out Sierra's upscaling algo used by those games.

If the games are totally broken they should at least have the `ADGF_UNSTABLE` flag.